cmd/buildtools/metadata.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"sort"
+
+	embeddedclusterv1beta1 "github.com/replicatedhq/embedded-cluster-kinds/apis/v1beta1"
+	"github.com/replicatedhq/embedded-cluster-kinds/types"
+	"github.com/replicatedhq/embedded-cluster/pkg/helm"
+	"github.com/replicatedhq/embedded-cluster/pkg/helpers"
+	"github.com/urfave/cli/v2"
+	"gopkg.in/yaml.v3"
+)
+
+var metadataCommand = &cli.Command{
+	Name:  "metadata",
+	Usage: "Perform operations on the version-metadata.json file",
+	Subcommands: []*cli.Command{
+		metadataListImagesCommand,
+		metadataExtractHelmChartImagesCommand,
+	},
+}
+
+var metadataListImagesCommand = &cli.Command{
+	Name:  "list-images",
+	Usage: "List images",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "metadata-path",
+			Usage: "Path to the metadata file",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		metadata, err := readMetadataFromFile(c.String("metadata-path"))
+		if err != nil {
+			return fmt.Errorf("failed to read metadata from file: %w", err)
+		}
+
+		for _, image := range metadata.Images {
+			fmt.Println(image)
+		}
+
+		return nil
+	},
+}
+
+var metadataExtractHelmChartImagesCommand = &cli.Command{
+	Name:  "extract-helm-chart-images",
+	Usage: "Extract images from Helm charts",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:  "metadata-path",
+			Usage: "Path to the metadata file",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		metadata, err := readMetadataFromFile(c.String("metadata-path"))
+		if err != nil {
+			return fmt.Errorf("failed to read metadata from file: %w", err)
+		}
+
+		repos := metadata.Configs.Repositories
+		charts := metadata.Configs.Charts
+		for _, chart := range metadata.BuiltinConfigs {
+			repos = append(repos, chart.Repositories...)
+			charts = append(charts, chart.Charts...)
+		}
+
+		images, err := extractImagesFromHelmExtensions(repos, charts)
+		if err != nil {
+			return fmt.Errorf("failed to extract images from helm extensions: %w", err)
+		}
+
+		for _, image := range images {
+			fmt.Println(image)
+		}
+
+		return nil
+	},
+}
+
+func readMetadataFromFile(path string) (*types.ReleaseMetadata, error) {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("read file: %w", err)
+	}
+
+	var metadata types.ReleaseMetadata
+	if err := json.Unmarshal(b, &metadata); err != nil {
+		return nil, fmt.Errorf("unmarshal metadata: %w", err)
+	}
+
+	return &metadata, nil
+}
+
+func extractImagesFromHelmExtensions(repos []embeddedclusterv1beta1.Repository, charts []embeddedclusterv1beta1.Chart) ([]string, error) {
+	hcli, err := helm.NewHelm(helm.HelmOptions{
+		K0sVersion: "1.29.6", // TODO
```

Comment:
can get this from `defaults.K0sVersion`.

Reply:
this is in buildtools so probably not

Reply:
ah in that case, maybe can use the `getK0sVersion` function?

Reply:
fixed to use the metadata